tests/e2e/spec/advanced.spec.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+import { browser, expect, $ } from '@wdio/globals';
+import {
+  enableExtension,
+  getExtensionElement,
+  setPrivacyToggle,
+  switchToPanel,
+} from '../utils.js';
+
+import { PAGE_DOMAIN, PAGE_URL } from '../wdio.conf.js';
+
+async function setCustomFilters(filters) {
+  await setPrivacyToggle('custom-filters', true);
+
+  const checkbox = await getExtensionElement(
+    'checkbox:custom-filters:trusted-scriptlets',
+  );
+
+  if (!(await checkbox.getProperty('checked'))) {
+    await checkbox.click();
+    await expect(checkbox).toHaveElementProperty('checked', true);
+  }
+
+  const input = await getExtensionElement('input:custom-filters');
+  await input.setValue(filters.join('\n'));
+
+  await getExtensionElement('button:custom-filters:save').click();
+
+  await expect(
+    getExtensionElement('component:custom-filters:result'),
+  ).toBeDisplayed();
+}
+
+describe('Advanced Features', function () {
+  before(enableExtension);
+
+  describe('Custom Filters', function () {
+    before(async function () {
+      await setCustomFilters([
+        `@@connect.facebook.net^`,
+        `${PAGE_DOMAIN}###custom-filter`,
+        `${PAGE_DOMAIN}##+js(rpnt, h1, Test Page, "Hello world")`,
+      ]);
+    });
+
+    after(() => setPrivacyToggle('custom-filters', false));
```

Comment:
probably best to update custom filters with empty input before disabling. just in case this state wont come up and mess other tests